src/components/decimal/decimal.test.tsx

```diff
@@ -0,0 +1,1406 @@
+    await user.keyboard("-");
+
+    expect(decimalInput).toHaveValue("-0.00");
+    expect(screen.getByTestId("hidden-input")).toHaveValue("-0.00");
+  });
+
+  it("allows the user to type a negative symbol at the end of the number", async () => {
+    const user = userEvent.setup();
+    render(<Decimal defaultValue="0.00" />);
+
+    const decimalInput = screen.getByRole("textbox");
+
+    decimalInput.focus();
+    await user.keyboard("{ArrowRight}");
+    await user.keyboard("-");
+
+    expect(decimalInput).toHaveValue("0.00-");
+    expect(screen.getByTestId("hidden-input")).toHaveValue("0.00-");
+  });
+
+  it.each([
+    [1, 1, ",234.56", ",234.56"],
+    [2, 2, "1234.56", "1234.56"],
+    [3, 3, "1,34.56", "134.56"],
+    [4, 4, "1,24.56", "124.56"],
+    [5, 5, "1,23.56", "123.56"],
+    [6, 6, "1,23456", "123456"],
+    [7, 7, "1,234.6", "1234.6"],
+    [0, 1, ",234.56", ",234.56"],
+    [0, 2, "234.56", "234.56"],
+    [0, 3, "34.56", "34.56"],
+    [0, 4, "4.56", "4.56"],
+    [0, 5, ".56", ".56"],
+    [0, 6, "56", "56"],
+    [0, 7, "6", "6"],
+    [0, 8, "", ""],
+    [1, 2, "1234.56", "1234.56"],
+    [1, 3, "134.56", "134.56"],
+    [1, 4, "14.56", "14.56"],
+    [1, 5, "1.56", "1.56"],
+    [1, 6, "156", "156"],
+    [1, 7, "16", "16"],
+    [1, 8, "1", "1"],
+    [2, 3, "1,34.56", "134.56"],
+    [2, 4, "1,4.56", "14.56"],
+    [2, 5, "1,.56", "1,.56"],
+    [2, 6, "1,56", "156"],
+    [2, 7, "1,6", "16"],
+    [2, 8, "1,", "1,"],
+    [3, 4, "1,24.56", "124.56"],
+    [3, 5, "1,2.56", "12.56"],
+    [3, 6, "1,256", "1256"],
+    [3, 7, "1,26", "126"],
+    [3, 8, "1,2", "12"],
+    [4, 5, "1,23.56", "123.56"],
+    [4, 6, "1,2356", "12356"],
+    [4, 7, "1,236", "1236"],
+    [4, 8, "1,23", "123"],
+    [5, 6, "1,23456", "123456"],
+    [5, 7, "1,2346", "12346"],
+    [5, 8, "1,234", "1234"],
+    [6, 7, "1,234.6", "1234.6"],
+    [6, 8, "1,234.", "1,234."],
+    [7, 8, "1,234.5", "1234.5"],
+  ])(
+    "allows deletion of content using the backspace key for a selection starting at index %s and ending at index %s",
+    async (selectionStart, selectionEnd, rawValue, sanitisedValue) => {
+      const user = userEvent.setup();
+      render(<Decimal defaultValue="1234.56" />);
+
+      const decimalInput = screen.getByRole("textbox") as HTMLInputElement;
+      decimalInput.focus();
+      decimalInput.setSelectionRange(selectionStart, selectionEnd);
+      await user.keyboard("{Backspace}");
+
+      expect(decimalInput).toHaveValue(rawValue);
+      expect(screen.getByTestId("hidden-input")).toHaveValue(sanitisedValue);
+    }
+  );
+
+  // as above, a simple cursor placement (no selection) at the start or end need special handling
+  it("doesn't delete anything when backspace is pressed with the cursor at the start of the input", async () => {
+    const user = userEvent.setup();
+    render(<Decimal defaultValue="1234.56" />);
+
+    const decimalInput = screen.getByRole("textbox") as HTMLInputElement;
```

Comment:
**suggestion:** I don't think you need the `as HTMLInputElement` here. I don't think you're doing anything that requires the input element type.

```suggestion
    const decimalInput = screen.getByRole("textbox");
```